From c728b4e89025e14c9207e0b762ad78b46040d746 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 6 Oct 2023 14:46:53 +0200
Subject: [PATCH 0563/1141] arm64: dts: st: change address of mlahb and m33
 node for the stm32mp251

Fix warning:
arch/arm64/boot/dts/st/stm32mp251.dtsi:2651.13-2671.4:
Warning (unit_address_vs_reg): /ahb: node has a reg or ranges property,
but no unit name

To clean DT warnings:
- provide an address 1 to the ahb node. the Soc node has the address 0,
- change the m33 address to 0. The current 0x0a040000 is not valid,
  it corresponds to the bsec mirror address.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: If77e0f1ddce90ef55c8852ee09f2beae283629aa
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/332087
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2690,16 +2690,16 @@
 		};
 	};
 
-	mlahb: ahb {
+	mlahb: ahb@1 {
 		compatible = "st,mlahb", "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
 		ranges = <0x0 0x0 0x0 0xfffffffc>;
 		dma-ranges = <0x0 0x0 0x0 0xfffffffc>;
 
-		m33_rproc: m33@a040000 {
+		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
-			reg = <0x0a040000 0x60000>;
+			reg = <0 0>;
 			resets = <&scmi_reset RST_SCMI_C2_R>,
 				 <&scmi_reset RST_SCMI_C2_HOLDBOOT_R>;
 			reset-names = "mcu_rst", "hold_boot";
